common/src/main/java/com/wynntils/core/config/OverlayGroupHolder.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.config;
+
+import com.google.gson.reflect.TypeToken;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.features.Feature;
+import com.wynntils.core.features.overlays.DynamicOverlay;
+import com.wynntils.core.features.overlays.Overlay;
+import com.wynntils.utils.ReflectionUtils;
+import java.lang.reflect.Field;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.reflect.FieldUtils;
+
+public class OverlayGroupHolder {
+    private final Field field;
+    private final Feature parent;
+    private final int defaultCount;
+
+    private final Class<?> overlayClass;
+
+    public OverlayGroupHolder(Field field, Feature parent, int defaultCount) {
+        this.field = field;
+        this.parent = parent;
+        this.defaultCount = defaultCount;
+
+        Type genericType = this.field.getGenericType();
+
+        if (genericType instanceof ParameterizedType parameterizedType) {
+            this.overlayClass =
+                    TypeToken.get(parameterizedType.getActualTypeArguments()[0]).getRawType();
+        } else {
+            throw new IllegalArgumentException("Field " + field.getName() + " is not a list.");
+        }
+    }
+
+    public String getConfigKey() {
+        return parent.getConfigJsonName() + ".groupedOverlay." + field.getName() + ".ids";
+    }
+
+    public int getOverlayCount() {
+        try {
+            return ((List<?>) FieldUtils.readField(field, parent, true)).size();
+        } catch (IllegalAccessException e) {
+            return defaultCount;
+        }
+    }
+
+    public List<? extends Overlay> getOverlays() {
+        try {
+            return ((List<? extends Overlay>) FieldUtils.readField(field, parent, true));
+        } catch (IllegalAccessException e) {
+            return List.of();
+        }
+    }
+
+    public int getDefaultCount() {
+        return defaultCount;
+    }
+
+    public Feature getParent() {
+        return parent;
+    }
+
+    public String getFieldName() {
+        return field.getName();
+    }
+
+    public Class<?> getOverlayClass() {
+        return overlayClass;
+    }
+
+    public void initGroup(List<Integer> ids) {
+        try {
+            List overlays = ReflectionUtils.createListWithType(overlayClass);
```

Comment:
The `overlays` List is untyped..? So why the ReflectionUtils wrapper? I assume this is something to do with casting for correct generics?

Reply:
Fixed it now. This is not even needed technically, I think, because of type-erasure.

Reply:
Yes, I don't think this does anything. With the cast in place, you can safely replace this call with just a new ArrayList().